src/api/apiRequest.ts

```diff
@@ -0,0 +1,40 @@
+import Axios, { AxiosRequestConfig } from 'axios';
+
+import { getIdToken } from './userApi';
+import { StatusCode } from '../utils/constants';
+import { setAxiosDefaults } from '../utils/axios-config';
+
+// Set axios defaults only once through the app's lifetime
+setAxiosDefaults();
+
+export const authenticatedApiRequest = async (axiosRequestConfig: AxiosRequestConfig) => {
+  try {
+    const idToken = await getIdToken();
+    axiosRequestConfig.headers = {
+      ...axiosRequestConfig.headers,
+      Authorization: `Bearer ${idToken}`,
+    };
+  } catch {
+    return { error: true };
+  }
+
+  return await apiRequest(axiosRequestConfig);
+};
+
+export const apiRequest = async (axiosRequestConfig: AxiosRequestConfig) => {
+  try {
+    const response = await Axios(axiosRequestConfig);
+
+    if (response.status === StatusCode.OK) {
+      return response.data;
```

Comment:
hva med status = Accepted og Created?

Reply:
eller er det tenkt at dette brukes kun i tilfeller hvor det skal returneres OK? ellers gjør vi det på "vanlig måte"?

Reply:
eller egne `apiRequest`-funksjoner?

Reply:
Alle responstyper kommer her etterhvert som vi utvider ja, men la bare inn OK nå siden de to kallene som bruker apiRequest for nå bare har det som respons